dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -297,5 +652,1699 @@ void setContextualProperty(String key, Object v) {
+            } else if (key == ATTACHMENTS) {
+                return attachments != NOT_FOUND;
+            } else if (key == ENCODING) {
+                return encoding != NOT_FOUND;
+            } else if (key == AbstractHTTPDestination.HTTP_CONTEXT) {
+                return httpContext != NOT_FOUND;
+            } else if (key == AbstractHTTPDestination.HTTP_CONFIG) {
+                return httpConfig != NOT_FOUND;
+            } else if (key == AbstractHTTPDestination.HTTP_CONTEXT_MATCH_STRATEGY) {
+                return httpContextMatchStrategy != NOT_FOUND;
+            } else if (key == HTTP_BASE_PATH) {
+                return httpBasePath != NOT_FOUND;
+            } else if (key == ASYNC_POST_RESPONSE_DISPATCH) {
+                return asyncPostDispatch != NOT_FOUND;
+            } else if (key == SECURITY_CONTEXT) {
+                return securityContext != NOT_FOUND;
+            } else if (key == AUTHORIZATION_POLICY) {
+                return authorizationPolicy != NOT_FOUND;
+            } else if (key == CERT_CONSTRAINTS) {
+                return certConstraints != NOT_FOUND;
+            } else if (key == AbstractHTTPDestination.SERVICE_REDIRECTION) {
+                return serviceRedirection != NOT_FOUND;
+            } else if (key == HTTP_SERVLET_RESPONSE) {
+                return httpServletResponse != NOT_FOUND;
+            } else if (key == RESOURCE_METHOD) {
+                return resourceMethod != NOT_FOUND;
+            } else if (key == ONE_WAY_REQUEST) {
+                return oneWayRequest != NOT_FOUND;
+            } else if (key == ASYNC_RESPONSE) {
+                return asyncResponse != NOT_FOUND;
+            } else if (key == THREAD_CONTEXT_SWITCHED) {
+                return threadContextSwitched != NOT_FOUND;
+            } else if (key == OutgoingChainInterceptor.CACHE_INPUT_PROPERTY) {
+                return cacheInputProperty != NOT_FOUND;
+            } else if (key == PhaseInterceptorChain.PREVIOUS_MESSAGE) {
+                return previousMessage != NOT_FOUND;
+            } else if (key == AbstractHTTPDestination.RESPONSE_HEADERS_COPIED) {
+                return responseHeadersCopied != NOT_FOUND;
+            } else if (key == SSE_EVENT_SINK) {
+                return sseEventSink != NOT_FOUND;
+            } else if (key == REQUESTOR_ROLE) {
+                return requestorRole != NOT_FOUND;
+            } else if (key == PARTIAL_RESPONSE_MESSAGE) {
+                return partialResponse != NOT_FOUND;
+            } else if (key == EMPTY_PARTIAL_RESPONSE_MESSAGE) {
+                return emptyPartialResponse != NOT_FOUND;
+            } else if (key == ENDPOINT_ADDRESS) {
+                return endpointAddress != NOT_FOUND;
+            } else if (key == INBOUND_MESSAGE) {
+                return inboundMessage != NOT_FOUND;
+            }
+        }
+        return super.containsKey(key);
+    }
+    @Override
+    public void putAll(Map<? extends String, ? extends Object> m) {
```

Comment:
Should make a copy of the Map and then as you set fields instead of calling the super way of putting all, you should, remove them from the copy of the Map and only send the subset to the super.putAll

Reply:
Excellent solution.